src/main/kotlin/org/jetbrains/research/ml/coding/assistant/dataset/model/DatasetRecord.kt

```diff
@@ -0,0 +1,77 @@
+package org.jetbrains.research.ml.coding.assistant.dataset.model
+
+import kotlinx.serialization.Serializable
+
+typealias Record = Map<String, String>
+
+/**
+ *  Model represents a single record entry in dataset
+ *  Model for a row in .csv file
+ */
+data class DatasetRecord(
+    val id: String,
+    val fragment: String,
+    val metaInfo: MetaInfo
+) {
+
+    internal constructor(record: Record) : this(
+        id = record[Keys.ID]!!,
+        fragment = record[Keys.FRAGMENT] ?: "",
+        metaInfo = MetaInfo(record)
+    )
+}
+
+/**
+ *  Model represents meta information attached to the dataset record
+ */
+@Serializable
+data class MetaInfo(
+    val age: Float?,
+    val programExperience: ProgramExperience?,
+    val testsResults: Double,
+    val task: DatasetTask
+) {
+    internal constructor(record: Record) : this(
+        age = record[Keys.AGE]?.toFloatOrNullWithDefault(-1.0f),
+        programExperience = record[Keys.PROGRAM_EXPERIENCE].toProgramExperience(),
+        testsResults = record[Keys.TESTS_RESULTS]!!.toDouble(),
+        task = DatasetTask.createFromString(record[Keys.TASK]!!)
```

Comment:
Please, don't use `!!`

Reply:
I think it would be great if you create a class with default values for each field or throw errors if the value is null. but please don't use `!!`

Reply:
yep, errors is a great choice

Reply:
I choose `NullPointerException` one. 😃